platform/core/src/classes/metadata/StudyMetadata.js

```diff
@@ -615,6 +615,21 @@ const makeDisplaySet = (series, instances) => {
     });
   }
 
+  // If MR series, sort the images by TemporalPositionIdentifier
+  if (imageSet.modality == 'MR') {
+    imageSet.sortBy((a, b) => {
+      // Sort by TemporalPositionIdentifier
+      if (a._instance.TemporalPositionIdentifier > b._instance.TemporalPositionIdentifier) return 1;
+      if (a._instance.TemporalPositionIdentifier < b._instance.TemporalPositionIdentifier) return -1;
+
+      // Sort by DiffusionBValue if defined (Diffusion series)
+      if (typeof(instance._instance.DiffusionBValue) != 'undefined') {
+        if (a._instance.DiffusionBValue > b._instance.DiffusionBValue) return 1;
+        if (a._instance.DiffusionBValue < b._instance.DiffusionBValue) return -1;
+      }
+    });
```

Comment:
The goal here being:

- If `TemporalPositionIdentifier` is available, sort by it.
- If one or both `TemporalPositionIdentifier`s are undefined, or if they are equal, defer to `DiffusionBValue`

Reply:
Yes! It makes me think:
 - in diffusion series `TemporalPositionIdentifier` would not be of any use (it is usually defined and equals 1)
- in non-diffusion series, `DiffusionBValue` is either `undefined` or defined AND equal to 0

There is another DICOM tag named `0020,0105 (NumberOfTemporalPositions)` that could prevent unnecessary sorting.

We could do something like: 
```
if (imageSet.modality === 'MR') {
  imageSet.sortBy((a, b) => {

    // Sort by TemporalPositionIdentifier
    if (instance._instance.NumberOfTemporalPositions !== 1) {      
      if (a._instance.TemporalPositionIdentifier > b._instance.TemporalPositionIdentifier) return 1;
      if (a._instance.TemporalPositionIdentifier < b._instance.TemporalPositionIdentifier) return -1;
    }

    // Sort by DiffusionBValue if defined (Diffusion series)
    if (typeof(instance._instance.DiffusionBValue) !== 'undefined') {
      if (a._instance.DiffusionBValue > b._instance.DiffusionBValue) return 1;
      if (a._instance.DiffusionBValue < b._instance.DiffusionBValue) return -1;
    }
  });
```

And in retrieveMetadata.js:
```
   // Get DiffusionBValue and TemporalPositionIdentifier for MR stack sorting
    DiffusionBValue: DICOMWeb.getNumber(instance['00189087']),
    TemporalPositionIdentifier: DICOMWeb.getNumber(instance['00200100']),
    NumberOfTemporalPositions: DICOMWeb.getNumber(instance['00200105']),
```

What do you think?